internal/cli/atlas/deployments/pause.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployments
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/setup"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
+)
+
+type PauseOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	options.DeploymentOpts
+	store  store.ClusterPauser
+	config setup.ProfileReader
+}
+
+var pauseTemplate = "Pausing deployment '{{.Name}}'.\n"
+
+var ErrDeploymentIsNotIDLE = errors.New("deployment state is not IDLE")
+
+func (opts *PauseOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *PauseOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if strings.EqualFold(opts.DeploymentType, options.LocalCluster) {
+		return opts.RunLocal(ctx)
+	}
+
+	return opts.RunAtlas()
+}
+
+func (opts *PauseOpts) RunLocal(ctx context.Context) error {
+	localDeployments, err := opts.GetLocalDeployments(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, deployment := range localDeployments {
+		if deployment.Name == opts.DeploymentName {
+			if err = opts.pauseContainer(ctx, deployment); err != nil {
+				return err
+			}
+
+			return opts.Print(
+				admin.AdvancedClusterDescription{
+					Name: &opts.DeploymentName,
+				})
+		}
+	}
+
+	return options.ErrDeploymentNotFound
+}
+
+func (opts *PauseOpts) pauseContainer(ctx context.Context, deployment options.Deployment) error {
+	if deployment.StateName == options.IdleState {
+		if _, err := opts.PodmanClient.StopContainers(ctx, opts.LocalMongotHostname()); err != nil {
```

Comment:
this might help for a "cleaner" mongot stop: https://github.com/mongodb/mongodb-atlas-cli/pull/2301/commits/b6c43a48c2ad37c2ab90ffe951f9e22fcc6e44f6

but let's see if the e2e is successful for this draft PR https://github.com/mongodb/mongodb-atlas-cli/pull/2301